src/sentry/mediators/sentry_app_installation_tokens/creator.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import
+
+from sentry.mediators import Mediator, Param
+from sentry.models import (
+    AuditLogEntryEvent, ApiToken, SentryAppInstallationToken
+)
+
+
+class Creator(Mediator):
+    sentry_app_installation = Param('sentry.models.SentryAppInstallation')
+    # analytics and audit params
+    user = Param('sentry.models.User')
+    request = Param('rest_framework.request.Request', required=False)
+
+    def call(self):
+        self.sentry_app = self.sentry_app_installation.sentry_app
```

Comment:
I think maybe to follow how we've done it the past, you could use `@memoize` https://github.com/getsentry/sentry/blob/master/src/sentry/mediators/sentry_app_installations/creator.py#L83